### PR TITLE
feature/backend

### PR DESCRIPTION
**Pull request by Rahaf91**

### CODE REVIEW

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Natural Cures Team,
your code looks very good already, well done! In some cases it might actually be overengineered for your given situation. Keep in mind to "keep it simple" whenever possible. There is an open question for me regarding your notes. You are using the _id in your frontend, but your notes don't have those defined in your schema.

File: db/models/Remedy.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  usage: { type: String },
+  videoUrlPreparation: { type: String },
+  category: { type: String },
+  isFavorite: { type: Boolean, default: false },
+  notes: [
+    {
+      text: { type: String, required: true },
+      timestamp: { type: Date, default: new Date().toLocaleString() },
+    },
+  ],
```
[the remark this answers is not in the document]
Comment (Rahaf91) [reply db/models/Remedy.js:19]: Yes, right anyway we will change it in the personalized date user sotry

**Comment by Rahaf91** [comment]
Thanks for reviewing! As for the notes, we initially assumed that MongoDB would automatically generate an _id for each note since it's part of an array within the document. Given that MongoDB typically generates unique _ids for documents, we thought the same would apply to array elements without needing to explicitly define the _id field in the schema.

File: components/Notes.js [hunk 2]
```diff
@@ -86,30 +91,30 @@ export default function Notes({
 
       <NoteList>
         {currentRemedy.notes?.map((note) => (
-          <NoteItem key={note.id}>
+          <NoteItem key={note._id}>
```
[the remark this answers is not in the document]
Comment (F-Kirchhoff) [reply components/Notes.js:94]: it only does this if the elements are documents in the collection, means that nested elements inside your remedy documents don't get _ids. You need to generate them by yourself.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Rahaf,
I added an explainer why the _ids are not there. Please create unique ids manually for your notes in the notes array.

**Comment by Rahaf91** [comment]
Sorry, maybe I didn't get your point, but when I look at the database, there is an _id for each note.
IMG_20241011_161539

**Comment by F-Kirchhoff** [comment]
ok I am sorry, I was not aware of this behavior, apparently when you use the $push functionality from mongoose these are created. Then your pr is good to go.

File: pages/api/remedies/[id]/notes/[noteId].js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        return;
+      }
+
+      response.status(200).json(updatedRemedy);
+      return;
+    } catch (error) {
+      response.status(500).json({ error: "Internal Server Error" });
+      return;
+    }
+  }
```
Comment (F-Kirchhoff) [inline pages/api/remedies/[id]/notes/[noteId].js:41]: This part should not be under the url notes/[noteId].js but notes/index.js instead. If you create a new note there is no noteId yet to call.

**Review by F-Kirchhoff — APPROVED** [review]
LGTM, you can proceed to QA 🚀

**Pull request merged**